include/commonpp/metric/reservoir/ExponentiallyDecaying.hpp

```diff
@@ -54,11 +51,18 @@ class ExponentiallyDecaying
 
     struct Sample
     {
-        Weight weight;
+        Weight mc_weight; //!< random weight for sampling multiplied with weight
+        Weight weight;    //!< time-based weight for exponential decay
         double value;
+
+        bool operator < (const Sample &o) const
```

Comment:
noexcept

Reply:
done. Should I do another PR?

Reply:
I merged a bit fast, if you don't mind but this is not really important :)

Reply:
will do. In the meanwhile: commonpp::thread::get_nb_physical_core() seems to be broken, returns 18446744073709551615 (0xFFFFFFFFUL) cores

Reply:
you built it without hwloc so it returns -1 maybe in this case an exception should be thrown

Reply:
finding hwloc seems to have worked until https://github.com/daedric/commonpp/commit/79d587b04beb1b6dc16f0e9059af70b38e05f3fe

it's also unfortunate that get_nb_physical_core() returns an int while the underlying type is a size_t. So the size_t(-1) gets promoted to a very large signed value and you can no longer check its return value against -1 or <0

Reply:
Ha, foudn the issue, will fix asap https://github.com/daedric/commonpp/commit/79d587b04beb1b6dc16f0e9059af70b38e05f3fe#diff-09f45435e63805480a54f356033f0cc9R44 
the unix/linux lib name has been broken. What way of returning the error would you suggest ?
I change the returned type to int and I let the caller check the number ?